src/liftdrag_plugin/advanced_liftdrag_plugin.cpp

```diff
@@ -0,0 +1,641 @@
+
+
+  /*
+  The Cm is divided in three sections: alpha>stall angle, alpha<-stall angle
+  -stall angle<alpha<stall angle. The Cm is assumed to be linear in the region between
+  -stall angle and stall angle, with the slope given by dCm/da. Once we get into the
+  stall regions, the Cm is still linear, but the slope changes to dCm_stall/da after stall
+  (also provided as an input). In the alpha>stall angle region the Cm is assumed to
+  always be positive or zero, in the alpha<-stall angle the Cm is assumed to always be
+  negative or zero.
+  */
+  double Cem;
+  if (this->alpha > this->alphaStall)
+  {
+    Cem = this->Cem0 + (this->Cema * this->alphaStall +
+          this->CemaStall * (this->alpha - this->alphaStall));
+  }
+  else if (this->alpha < -this->alphaStall)
+  {
+    Cem = this->Cem0 + (-this->Cema * this->alphaStall +
+          this->CemaStall * (this->alpha + this->alphaStall));
+  }
+  else{
+    Cem = this->Cem0 + this->Cema * this->alpha;
+  }
+  // Add sideslip effect, if any
+  Cem = this->Cemb * this->beta;
+  // Add rate terms
+  Cem = Cem + this->Cemp*p + this->Cemq*q + this->Cemr * r;
+
+  // Take into account the effect of control surface deflection angle to Cm
+  Cem += Cem_ctrl_tot;
+  gzdbg << "Current Cm:" << Cem << "\n";
+
+  // Compute roll moment coefficient, Cell
+  // Start with angle of attack, sideslip, and control terms
+  double Cell = this-> Cella * this->alpha + this->Cellb * this-> beta + Cell_ctrl_tot;
+  // Add rate terms
+  Cell = Cell + this->Cellp*p + this->Cellq*q + this->Cellr * r;
+
+  // Compute yaw moment coefficient, Cen
+  // Start with angle of attack, sideslip, and control terms
+  double Cen = this->Cena * this->alpha + this->Cenb * this->beta + Cen_ctrl_tot;
+  // Add rate terms
+  Cen = Cen + this->Cenp*p + this->Cenq*q + this->Cenr * r;
+  gzdbg << "Current Cl:" << Cell << "\n";
+  gzdbg << "Current Cn:" << Cen << "\n";
+
+  // Compute moment (torque)
+  ignition::math::Vector3d moment = (Cem * dyn_pres * this->area * mac * bodyY) + (Cell * dyn_pres * this->area * span * bodyX) + (Cen * dyn_pres * this->area * span * bodyZ);
+
+  // compute force about cg in inertial frame
+  ignition::math::Vector3d force = lift + drag + sideforce;
+
+    gzdbg << "force: " << force << "\n";
+    gzdbg << "moment: " << moment << "\n\n";
+
+  // Correct for nan or inf
+  force.Correct();
+  this->ref_pt.Correct();
+  moment.Correct();
+
+  // Apply forces at ref_pt (with torques for position shift)
+  this->link->AddForceAtRelativePosition(force, this->ref_pt);
+  this->link->AddTorque(moment);
+
+  auto relative_center = this->link->RelativePose().Pos() + this->ref_pt;
+
+  // Publish force and center of pressure for potential visual plugin.
+  // - dt is used to control the rate at which the force is published
+  // - it only gets published if 'topic_name' is defined in the sdf
+  if (dt > 1.0 / 10 && this->sdf->HasElement("topic_name")) {
```

Comment:
could you make `1.0 / 10` a named const expr so we reduce "magic numbers"? e.g. 
```cpp
static const expr kForceVisualizationPublishingInterval = 0.1; // [sec]
```

Reply:
Good idea. Where, exactly, would I place this? I want to make sure I'm not putting it in the wrong file.